public/scripts/project/metadata.js

```diff
@@ -127,17 +151,50 @@ define(function(require) {
       url: url
     });
     request.done(function(data) {
-      setMetadata(root, data, callback);
+      callback(data);
     });
     request.fail(function(jqXHR, status, err) {
       callback(err);
     });
   }
 
+  function loadAnonymousMetadata(config, callback) {
+    callback = lockSafeCallback(callback);
+
+    lock(function() {
```

Comment:
Same deal here when you rebase, make sure to wrap this callback like I do with others.